pkg/k8shandler/types.go

```diff
@@ -0,0 +1,48 @@
+package k8shandler
+
+import (
+	"fmt"
+
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
+	"github.com/openshift/cluster-logging-operator/pkg/utils"
+	sdk "github.com/operator-framework/operator-sdk/pkg/sdk"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ClusterLogging struct {
+	*logging.ClusterLogging
+}
+
+//NewClusterLogging wraps the public api types into an instance upon whichj
+//we can call attached methods
+func NewClusterLogging(api *logging.ClusterLogging) *ClusterLogging {
+	return &ClusterLogging{api}
+}
+
+//addOwnerRefTo adds an ownerRef from this ClusterLogging instance to the given object
+func (clusterLogging *ClusterLogging) addOwnerRefTo(object metav1.Object) {
+	ownerRef := utils.AsOwner(clusterLogging.ClusterLogging)
+	utils.AddOwnerRefToObject(object, ownerRef)
+}
+
+//DoesClusterLoggingExist determines if the specified cluster exists
+func (clusterLogging *ClusterLogging) DoesClusterLoggingExist() bool {
+	exists, _ := utils.DoesClusterLoggingExist(clusterLogging.ClusterLogging)
+	return exists
+}
+
+//createOrUpdateServiceAccount creates or updates a ServiceAccount for logging with the given name
+func (logging *ClusterLogging) createOrUpdateServiceAccount(name string) error {
```

Comment:
I would argue this belongs in `utils` and the functions that return an object definition would belong in here.